packages/web-storage/src/helpers.ts

```diff
@@ -0,0 +1,210 @@
+import cookie from 'js-cookie';
+
+import localStorageInstance from './core';
+
+import { BUCKET_SEPERATOR, BUCKET_UNAVAILABLE_ERROR } from './constants';
+
+/**
+ * Get Full Key for an Item that is to be Stored.
+ *
+ * @remarks
+ * This method is a part of our Storage Library.
+ *
+ * @param key - Key which is passed from frontend.
+ * @param bucket - Bucket where data is to be stored.
+ *
+ * @returns a plain key that seperates bucket and initial key provided by a seperator.
+ *
+ * @example
+ * Here's an example how key is generated:
+ * ```
+ * console.log(getFullKeyForItem('kjsdfgkj','PERSISTED')); => Output will be 'PERSISTED$$%&kjsdfgkj'
+ * ```
+ *
+ * @internal
+ */
+
+
+export const getFullKeyForItem = (key: string, bucket: string): string => {
+  return `${bucket} + ${BUCKET_SEPERATOR} + ${key}`;
+};
+
+/**
+ * Get Bucket Name of a key
+ *
+ * @remarks
+ * This method is a part of our Storage Library.
+ *
+ * @param key - Key which was finally generated to store data.
+ *
+ * @returns bucket name.
+ *
+ * @example
+ * Here's an example how bucket name is returned
+ * ```
+ * console.log(getBucketNameFromKey('PERSISTED$$%&kjsdfgkj')); => Output will be 'PERISTED'
+ * ```
+ * ```
+ * console.log(getBucketNameFromKey('someKeyNotHavingOurSeperator')); => Output will be 'No Bucket Associated!'.
+ * ```
+ * @internal
+ */
+
+export const getBucketNameFromKey = (key: string): string => {
+  if (key.includes(BUCKET_SEPERATOR)) {
+    return key.split(BUCKET_SEPERATOR)[0];
+
+  } else {
+    return BUCKET_UNAVAILABLE_ERROR;
+  }
+};
+
+/**
+ * Get User Provided Key from our Stored Key
+ *
+ * @remarks
+ * This method is a part of our Storage Library.
+ *
+ * @param key - Key which was finally generated to store data.
+ *
+ * @returns user provided key.
+ *
+ * @example
+ * Here's an example how user key is returned
+ * ```
+ * console.log(getUserKeyFromStoredKey('PERSISTED$$%&kjsdfgkj')); => Output will be 'kjsdfgkj'
+ * ```
+ * ```
+ * console.log(getUserKeyFromStoredKey('someKeyNotHavingOurSeperator')); => Output will be 'someKeyNotHavingOurSeperator'.
+ * ```
+ * @internal
+ */
+
+export const getUserKeyFromStoredKey = (key: string): string => {
+  if (key.includes(BUCKET_SEPERATOR)) {
+    return key.split(BUCKET_SEPERATOR)[1];
+
+  } else {
+    return key;
+  }
+};
+
+
+/**
+ * Gets a Particular Bucket Size in Local Storage
+ *
+ * @remarks
+ * This method is a part of our Storage Library.
+ *
+ * @param bucket - Bucket where data is to be stored.
+ *
+ * @returns Bucket Size in KB.
+ * @example
+ * Here's an example how user key is returned
+ * ```
+ * console.log(calculateBucketSizeInLocalStorage('PERSISTED')); => Output will be '40.00'
+ * ```
+ * @internal
+ *
+ */
+
+export const calculateBucketSizeInLocalStorage = (bucket: string) : string | null => {
+  if (typeof (Storage) !== 'undefined') {
+    let data = '';
+
+    for (let index = 0; index < localStorage.length; index++) {
```

Comment:
Let's discuss this iteration on office. 
Let's see if we can use different data structures